content/applications/finance/accounting/others/inventory/avg_price_valuation.rst

```diff
@@ -1,167 +1,477 @@
+
+   #. **Inventory value** increases because more 4 tables with an average cost of `$12` each enters
+      the warehouse as an asset.
+   #. Conversely, **credit** the **Stock Input** account `$64`.
+
+#. Like #2, when the bill for 4 tables is received,
+
+   #. **Stock input** account is debited, which reconciles it to `$0`.
+   #. On the opposite end, **Accounts Payable** is credited, and the `$64` is handed off to the
+      accountants to pay the vendor bill.
+
+Product delivery
+****************
+
+#. When 10 products are delivered, the **Stock Input** account is untouched because there are no new
+   products coming in.
+   #. **Inventory valuation** credited `$120`
+   #. Sales recorded in **Accounts Receivable**.
+
+Balance accounts on product return
+**********************************
+
+#. Return 1 table to the vendor who sold it at `$10`
+
+=======
+
+Explaining journal entries above
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. When 8 tables arrive at the warehouse, :ref:`AVCO inventory valuation
+   <inventory/avg_cost/math_table>` is calculated on the incoming products.
+
+   #. Inventory value increases by `$80`, as it reflects that the company currently has `$80` worth
+      of tables in stock.
+   #. `$80` is set aside in the *stock input* account, since this amount is owed to the vendor.
+   #. Credit the *stock input* account `$80`.
+   #. Debit the *Inventory Valuation* account `$80`.
+
+#. Bill sent by vendor for the 8 tables is received.
+
+   #. Use `$80` in the *stock input* account to pay the bill. This cancels out and the account now
+      holds `$0`.
+   #. Debit the *stock input* account `$80`.
+   #. Credit the *Account payable* `$80`. This account stores the amount the company owes others, so
+      accountants use the amount to write checks to vendors.
+
+#. When 4 tables arrive at the warehouse, the inventory valuation is recalculated using the methods
+   explained :ref:`previously<inventory/avg_cost/math_table>`.
+
+   #. The *stock input* account stores the `$64` owed to the vendor. The amount in this account is
+      unrelated to the inventory value, because the amount the table is valued at within the company
+      is separate from the price the vendor sold the product for.
+   #. Credit *stock input* account `$64`.
+
+#. Similarly to #2, when the bill for 4 tables is received,
+   #. *Stock input* account is debited, which reconciles it to `$0`.
+   #. On the opposite end, *Account payable* is credited, and the `$64` is handed off to the
+      accountants to pay the vendor bill.
+
+#. When 10 products are delivered, the *stock input* account is untouched because there are no new
+   products coming in.
+   #. Inventory valuation credited `$120`
+
+#. Return 1 table to the vendor who sold it at `$10`
+>>>>>>> 9363ec06a ([IMP]: flesh out accounting and math)
+   #. Debit *stock input* account `$12` (Why?)
+      #. also how to account for the `$2` price difference?
```

Comment:
The *Stock Input* account is debited $12 because the avg cost of a table is 12 at the moment the product is returned (even though it was bought for $10). The $2 price difference has to be accounted for in the *Price Difference Account* field under the product category. The *Price Difference Account* differs from localization to localization of course :).

Note that this only works with automated inventory valuation, and it could be interesting to add as note (+ :doc: or :ref: to this page https://www.odoo.com/documentation/16.0/applications/inventory_and_mrp/inventory/management/reporting/inventory_valuation_config.html?highlight=inventory%20valuation)